src/main/java/bio/terra/service/resourcemanagement/azure/AzureAuthService.java

```diff
@@ -21,32 +23,59 @@ public AzureAuthService(AzureResourceConfiguration configuration) {
+   * @param storageAccountResource The storage account that DataLake client should be built from
    * @return an authenticated DataLake client
    */
   public DataLakeServiceClient getDataLakeClient(
       BillingProfileModel profileModel, AzureStorageAccountResource storageAccountResource) {
-    AzureResourceManager client =
-        configuration.getClient(UUID.fromString(profileModel.getSubscriptionId()));
+    String key = getStorageAccountKey(profileModel, storageAccountResource);
+
+    // Create a data lake client by authenticating using the found key
+    return new DataLakeServiceClientBuilder()
+        .credential(new StorageSharedKeyCredential(storageAccountResource.getName(), key))
+        .endpoint("https://" + storageAccountResource.getName() + ".dfs.core.windows.net")
+        .buildClient();
+  }
 
+  /**
+   * Return an authenticated {@link BlobContainerClient} client using key-based authentication
+   *
+   * @param profileModel The object containing user tenant information
+   * @param storageAccountResource The sa that BlobContainerClient client should be built from
+   * @param containerName The name of the container to create the client for
+   * @return an authenticated {@link BlobContainerClient}
+   */
+  public BlobContainerClient getBlobContainerClient(
+      BillingProfileModel profileModel,
+      AzureStorageAccountResource storageAccountResource,
+      String containerName) {
     // Obtain a secret key for the associated storage account
-    String key =
-        client
-            .storageAccounts()
-            .getByResourceGroup(
-                storageAccountResource.getApplicationResource().getAzureResourceGroupName(),
-                storageAccountResource.getName())
-            .getKeys()
-            .get(0)
-            .value();
+    String key = getStorageAccountKey(profileModel, storageAccountResource);
 
     // Create a data lake client by authenticating using the found key
-    return new DataLakeServiceClientBuilder()
+    return new BlobContainerClientBuilder()
         .credential(new StorageSharedKeyCredential(storageAccountResource.getName(), key))
         .endpoint("https://" + storageAccountResource.getName() + ".blob.core.windows.net")
+        .containerName(containerName)
         .buildClient();
   }
+
+  /** Obtain a secret key for the associated storage account */
+  private String getStorageAccountKey(
+      BillingProfileModel profileModel, AzureStorageAccountResource storageAccountResource) {
+    AzureResourceManager client =
+        configuration.getClient(UUID.fromString(profileModel.getSubscriptionId()));
+
+    return client
+        .storageAccounts()
+        .getByResourceGroup(
+            storageAccountResource.getApplicationResource().getAzureResourceGroupName(),
+            storageAccountResource.getName())
+        .getKeys()
+        .get(0)
+        .value();
+  }
```

Comment:
It's interesting that we don't pull this key from vault

Reply:
This is actually a key that Azure stores for us...it's basically a property of the storage account